packages/mcp/src/__tests__/evalWithOpenAI.test.ts

```diff
@@ -0,0 +1,217 @@
+    }
+    return PrimerMcpClient.instance
+  }
+
+  async listTools() {
+    return await this.client.listTools()
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async callTool({name, args}: {name: string; args: any}) {
+    return await this.client.callTool({name, arguments: args})
+  }
+
+  getClient() {
+    return this.client
+  }
+
+  /** Gracefully disconnect transport/client and reset singleton */
+  async teardown() {
+    await this.client.close()
+
+    PrimerMcpClient.instance = null
+  }
+}
+
+const mcp = await PrimerMcpClient.getInstance()
+const openai = new OpenAI({apiKey: process.env.OPEN_AI_KEY})
+
+async function runMcpOpenAiWithPrompt(prompt: string) {
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  const {tools} = await mcp.listTools()
+  const toolUsage: Record<string, number> = {}
+  let aiResponse: string | null = null
+
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  const openAITools = tools.map((tool: any) => ({
+    type: 'function' as const,
+    function: {
+      name: tool.name,
+      description: tool.description,
+      parameters: tool.inputSchema,
+    },
+  }))
+
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  const messages: any[] = [{role: 'user', content: prompt}]
+
+  // Initial OpenAI completion
+  const completion = await openai.chat.completions.create({
+    model: GPT_MODEL,
+    messages,
+    tools: openAITools,
+    // eslint-disable-next-line camelcase
+    tool_choice: 'auto',
+  })
+
+  // Handle tool calls if present
+  const choice = completion.choices[0]
+
+  if (choice.finish_reason === 'tool_calls') {
+    messages.push({
+      role: 'assistant',
+      content: choice.message.content ?? '',
+      refusal: choice.message.refusal ?? '',
+      // eslint-disable-next-line camelcase
+      tool_calls: choice.message.tool_calls ?? [],
+    })
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+    const toolCalls: any = choice.message.tool_calls || []
+    for (const toolCall of toolCalls) {
+      const tool = toolCall.function
+      if (tool && tool.name && tool.arguments) {
+        toolUsage[tool.name] = (toolUsage[tool.name] || 0) + 1
+        const toolResult = await mcp.callTool({
+          name: tool.name,
+          args: JSON.parse(tool.arguments),
+        })
+        messages.push({
+          role: 'tool',
+          // eslint-disable-next-line camelcase
+          tool_call_id: toolCall.id,
+          content: JSON.stringify(toolResult),
+        })
+      }
+    }
+
+    const finalResponse = await openai.chat.completions.create({
+      model: GPT_MODEL,
+      messages,
+    })
+
+    aiResponse = finalResponse.choices[0].message.content
+  } else {
+    // Handle non-tool call response
+    aiResponse = choice.message.content
+  }
+
+  return {
+    toolUsage,
+    aiResponse,
```

Comment:
Returns the number of times a Primer MCP tool was called and the AIResponse to the prompt.